src/distance/ConfigurationRanker.cpp

```diff
@@ -0,0 +1,71 @@
+#include "aikido/distance/ConfigurationRanker.hpp"
+
+namespace aikido {
+namespace distance {
+
+using statespace::dart::ConstMetaSkeletonStateSpacePtr;
+using ::dart::dynamics::ConstMetaSkeletonPtr;
+
+//==============================================================================
+ConfigurationRanker::ConfigurationRanker(
+    ConstMetaSkeletonStateSpacePtr metaSkeletonStateSpace,
+    ConstMetaSkeletonPtr metaSkeleton)
+  : mMetaSkeletonStateSpace(std::move(metaSkeletonStateSpace))
+  , mMetaSkeleton(std::move(metaSkeleton))
+{
+  if (!mMetaSkeletonStateSpace)
+    throw std::invalid_argument("MetaSkeletonStateSpace is nullptr.");
+
+  if (!mMetaSkeleton)
+    throw std::invalid_argument("MetaSkeleton is nullptr.");
+
+  mDistanceMetric = createDistanceMetricFor(
+      std::dynamic_pointer_cast<statespace::CartesianProduct>(
+          std::const_pointer_cast<statespace::dart::MetaSkeletonStateSpace>(
+              mMetaSkeletonStateSpace)));
+}
+
+//==============================================================================
+statespace::ConstStateSpacePtr ConfigurationRanker::getStateSpace() const
+{
+  return mMetaSkeletonStateSpace;
+}
+
+//==============================================================================
+void ConfigurationRanker::rankConfigurations(
+    std::vector<statespace::CartesianProduct::State*>& configurations)
+{
+  std::vector<std::pair<statespace::CartesianProduct::State*, double>>
+      scoredConfigurations(configurations.size());
+  for (std::size_t i = 0; i < configurations.size(); ++i)
+  {
+    scoredConfigurations[i].first = configurations[i];
+    scoredConfigurations[i].second = evaluateConfiguration(configurations[i]);
+  }
+
+  struct sortingFunction
```

Comment:
Can we move this out of this method into an anonymous namespace? Also, maybe a more descriptive name would be helpful.